src/main/java/org/jitsi/videobridge/AbstractEndpoint.java

```diff
@@ -74,6 +91,14 @@
      */
     private boolean expired = false;
 
+    /**
+     * The maximum set of constraints applied by all receivers of this endpoint
+     * in the conference. The client needs to send _at least_ this to satisfy
+     * all receivers.
+     */
+    private VideoConstraints maxReceiverVideoConstraints
+        = new VideoConstraints(BitrateControllerConfig.Config.onstageIdealHeightPx());
```

Comment:
This starts as `onstageIdealHeightPx`,  but then defaults to `defaultMaxReceiverVideoConstraints = thumbnailVideoConstraints` (which is different). Is that on puspose?

Reply:
If I understand correctly, the expectation the current client code has is that endpoints that are not specified (either pinned or selected) default to the lowest layer. If we want to allow senders to completely stop sending video in the future, should we handle this elsewhere (i.e. have receivers add themselves with a default of `thumbnailVideoConstraints`) and default to `none` here?

Reply:
> If we want to allow senders to completely stop sending video in the future

I'm thinking we can augment the video constraints with a max with the semantics: "The bridge will not exceed this resolution. If there aren't any encodings does provide resolution or less, the stream from this endpoint becomes suspended."

Much like in the webrtc constraints spec.

Reply:
Something like this 592b92e4eab21a6f7a1d2027411cd617352d3c4f (nothing uses this atm and I should probably revert this, unless you think it's useful).